ddtrace/profiling/collector/_lock.py

```diff
@@ -261,6 +261,13 @@ def _get_lock_call_loc_with_name(self) -> typing.Optional[str]:
             # 2: acquire/release (or __enter__/__exit__)
             # 3: caller frame
             # And we expect additional frame if WRAPT_C_EXT is False
+            if __debug__:
```

Comment:
Does our build system properly set debug vs release for CI/release builds?

Reply:
Changed to use `DD_PROFILING_ENABLE_ASSERTS`